src/nvim/api/buffer.c

```diff
@@ -1465,15 +1465,21 @@ Integer nvim_buf_set_extmark(Buffer buffer, Integer ns_id,
   }
 
   if (col2 >= 0) {
-    if (line2 >= 0) {
-      len = STRLEN(ml_get_buf(buf, (linenr_T)line2+1, false));
+    if (line2 >= 0 && line2 != buf->b_ml.ml_line_count) {
```

Comment:
this looks weird to me, shouldn't be `0 <= line2 && line2 < buf->b_ml.ml_line_count` ?